FrontMen/fm-hours
# Feature/email reminder

# Changes

Refactor to email.ts to create email by passing content, so there's a template of sorts that take content and subject.
Ability to send email reminder per timesheet as an admin.
Records now has a subpage records/:startdate to show a specific 
From /timesheets you can trigger an email for every employee with an empty or denied timesheet based on selected week (start date)

## Related issues

Closes https://github.com/FrontMen/fm-hours/issues/203

## Added
A reminder email

## How to test
1. Go to timesheets
2. Click on a pending/denied/new/empty timesheet (ideally your own)
3. Note a reminder button
4. Click it (if the timesheet is yours you'll receive an email)
5. Note email reminder. Click on link
6. Should load the correct week

1. Go to timesheets
2. Weeks in dropdown match headers in table
3. Not sure how you can test sending cos you will spam everybody :)) maybe approve a week for everyone but you and send a reminder for that week then check only you got an email.

## Screenshots
empty
![image](https://user-images.githubusercontent.com/85111889/127684458-0a040653-b3b1-47d3-a697-bb06ca6ed748.png)
denied
![image](https://user-images.githubusercontent.com/85111889/127684503-48da9588-dce6-467c-9dbe-a8dbc65ae7ed.png)
new / pending
![image](https://user-images.githubusercontent.com/85111889/127684544-daaf8c58-dbad-4e68-8f41-a4287e8de5ec.png)
approved
![image](https://user-images.githubusercontent.com/85111889/127684633-ba7bce9d-5558-4e72-911d-9d049be259ae.png)
email 
![image](https://user-images.githubusercontent.com/85111889/127684686-df10f4ae-4ae6-456a-8ec8-1a363d11a879.png)
Timesheets page 
![image](https://user-images.githubusercontent.com/85111889/127888625-d5199f20-9307-474f-a2e6-7b5fd3ed4f27.png)
